libraries/default.rb

```diff
@@ -329,23 +336,25 @@ def git_repo_url(git_data)
 
     def get_elk_info
       stack_name = stack_shortname
-      ::Chef::Mixin::DeepMerge.deep_merge({
```

Comment:
Reworked many of the version settings here and added apt repo urls to facilitate easier pinning. In addition, I switched to using string hash keys (vs symbols) after discovering that the merging of default values and cluster config overrides was actually working (!).